src/assignment4/Assignment4.java

```diff
@@ -2,6 +2,14 @@
 
 public class Assignment4 {
 
+    /**
+     * Makes a copy of the map (boolean[][]) used by the functions that destroy the map while processing it.
+     * 
+     * @param map

```

Comment:
If you include `@param` comments, please add descriptions of parameters. If you think they are self-explanatory, please remove `@param`s as they do not bring any value.